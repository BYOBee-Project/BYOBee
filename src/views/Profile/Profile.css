.profile {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.profile-title {
  padding: 0;
  margin: 8vh 0 0 0;
}

.profile-cards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  width: 80vw;
  gap: 2vw;
  margin-top: 4vh;
  padding-bottom: 4vh;
}
