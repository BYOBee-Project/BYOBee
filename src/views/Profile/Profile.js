import React from 'react';
import './Profile.css';
import { logout } from '../../services/users';
import { getUserBees } from '../../services/submissions';
import { useState, useEffect } from 'react';
import BeeCard from '../../components/BeeCard/BeeCard';
// import { findBeeById } from '../../services/bees';

export default function Profile({ setCurrentUser, currentUser }) {
  const [userBees, setUserBees] = useState([]);
  // const [stockBee, setStockBee] = useState({});
  const [loading, setLoading] = useState(true);

  const id = currentUser.user.id;

  const logoutUser = async () => {
    await logout();
    setCurrentUser(null);
  };

  useEffect(() => {
    const fetchData = async () => {
      const data = await getUserBees(id);
      setUserBees(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  // useEffect(() => {
  //   const fetchData = async () => {
  //     const data = await findBeeById(id);
  //     // console.log(data);
  //   };
  //   fetchData();
  // }, [id]);

  if (loading) return <h1>Loading...</h1>;

  return (
    <div className="profile">
      {userBees.map((bee) => (
        <BeeCard key={bee.id} bee={bee} />
      ))}

      <button onClick={logoutUser}>Logout</button>
    </div>
  );
}
