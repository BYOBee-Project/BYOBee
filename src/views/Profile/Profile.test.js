import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Profile from './Profile.js';

test.skip('Profile view should render the Profile component', async () => {
  const container = render(
    <MemoryRouter>
      <Route>
        <Profile currentUser={{ user: { id: 'd1beb9ec-d137-4e7f-ac5c-4c9197075a15' } }} />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('Bee Collection');
  expect(container).toMatchSnapshot();
});
