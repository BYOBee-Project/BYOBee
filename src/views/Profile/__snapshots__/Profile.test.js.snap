// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile view should render the Profile component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile"
      >
        <h1
          class="profile-title"
        >
          Bee Collection
        </h1>
        <div
          class="profile-cards"
        >
          <div
            class="user-card"
          >
            <h3
              class="user-card-title"
            >
              Black-notched Bumblebee
            </h3>
            <a
              href="/profile/32"
            >
              <img
                class="user-card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/d1beb9ec-d137-4e7f-ac5c-4c9197075a15/3/2022-01-20T20:39:38.457Z.jpg"
              />
            </a>
            <div
              class="user-card-buttons"
            >
              <button>
                <a
                  href="/profile/32"
                >
                  View
                </a>
              </button>
              <button>
                <a
                  href="/edit/32"
                >
                  Edit
                </a>
              </button>
              <button
                class="delete-button"
              >
                x
              </button>
            </div>
          </div>
        </div>
        <button>
          Logout
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile"
    >
      <h1
        class="profile-title"
      >
        Bee Collection
      </h1>
      <div
        class="profile-cards"
      >
        <div
          class="user-card"
        >
          <h3
            class="user-card-title"
          >
            Black-notched Bumblebee
          </h3>
          <a
            href="/profile/32"
          >
            <img
              class="user-card-image"
              src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/d1beb9ec-d137-4e7f-ac5c-4c9197075a15/3/2022-01-20T20:39:38.457Z.jpg"
            />
          </a>
          <div
            class="user-card-buttons"
          >
            <button>
              <a
                href="/profile/32"
              >
                View
              </a>
            </button>
            <button>
              <a
                href="/edit/32"
              >
                Edit
              </a>
            </button>
            <button
              class="delete-button"
            >
              x
            </button>
          </div>
        </div>
      </div>
      <button>
        Logout
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
